runtime/account/src/account.rs

```diff
@@ -0,0 +1,156 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The AccountId20 implementation.
+//!
+//! It includes AccountId conversations
+//!
+//! Ex:
+//! use std::str::FromStr;
+//! AccountId20::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap()
+
+use sp_core::{H160};
+use sp_std::hash::Hash;
+use codec::{Encode, Decode};
+#[cfg(feature = "std")]
+use sp_core::hexdisplay::HexDisplay;
+
+#[cfg(feature = "std")]
+use std::str::FromStr;
+
+#[cfg(feature = "full_crypto")]
+use sp_core::crypto::{PublicError};
+
+use pallet_evm::AddressMapping;
+
+/// An opaque 20-byte cryptographic identifier.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Default, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Hash))]
+pub struct AccountId20([u8; 20]);
+
+#[cfg(feature = "std")]
+impl FromStr for AccountId20 {
+	type Err = PublicError;
+	fn from_str(s: &str) -> Result<Self, PublicError> {
+		let hex_without_prefix = s.trim_start_matches("0x");
+		if hex_without_prefix.len() == 40 {
+			let mut bytes = [0u8; 20];
+			hex::decode_to_slice(hex_without_prefix, &mut bytes)
+				.map_err(|_| PublicError::InvalidFormat)
+				.map(|_| Self::from(bytes))
+		} else {
+			Err(PublicError::BadLength)
+		}
+	}
+}
+
+impl AsRef<[u8]> for AccountId20 {
+	fn as_ref(&self) -> &[u8] {
+		&self.0[..]
+	}
+}
+
+impl AsMut<[u8]> for AccountId20 {
+	fn as_mut(&mut self) -> &mut [u8] {
+		&mut self.0[..]
+	}
+}
+
+impl AsRef<[u8; 20]> for AccountId20 {
+	fn as_ref(&self) -> &[u8; 20] {
+		&self.0
+	}
+}
+
+impl AsMut<[u8; 20]> for AccountId20 {
+	fn as_mut(&mut self) -> &mut [u8; 20] {
+		&mut self.0
+	}
+}
+
+impl From<[u8; 20]> for AccountId20 {
+	fn from(x: [u8; 20]) -> AccountId20 {
+		AccountId20(x)
+	}
+}
+
+impl From<H160> for AccountId20 {
+	fn from(x: H160) -> AccountId20 {
+		AccountId20(x.0)
+	}
+}
+
+impl<'a> sp_std::convert::TryFrom<&'a [u8]> for AccountId20 {
+	type Error = ();
+	fn try_from(x: &'a [u8]) -> Result<AccountId20, ()> {
+		if x.len() == 20 {
+			let mut r = AccountId20::default();
+			r.0.copy_from_slice(x);
+			Ok(r)
+		} else {
+			Err(())
+		}
+	}
+}
+
+impl From<AccountId20> for [u8; 20] {
+	fn from(x: AccountId20) -> [u8; 20] {
+		x.0
+	}
+}
+
+/// Identity address mapping, required for EVM address mapping
+pub struct IdentityAddressMapping;
+
+impl AddressMapping<AccountId20> for IdentityAddressMapping {
+	fn into_account_id(address: H160) -> AccountId20 {
+		address.to_fixed_bytes().into()
+	}
+}
```

Comment:
Why is this included here? Isn't it already in pallet-evm? https://github.com/paritytech/substrate/blob/rococo-branch/frame/evm/src/lib.rs#L160

Reply:
Not sure how I'm supposed to do it in that case.
The one defined in the evm is for the H160 structure, while we are using a `AccountId20 ([u8; 20])`.
Is implementing the To<H160> the solution for it ?